recipe/meta.yaml

```diff
@@ -1,64 +1,125 @@
+    - urllib3 >=1.25.4,<1.27
+
   run:
-    - python
-#    - python >=3.7
-    - botocore =={{ botocore_pin }}
-    - colorama >=0.2.5,<0.4.5
-    - docutils >=0.10,<0.17
-    - rsa >=3.1.2,<4.8
-    - s3transfer >=0.6.0,<0.7.0
-    - pyyaml >=3.10,<5.5
+    - python >=3.8
+    # Commented lines are the original requirements from
+    # https://github.com/aws/aws-cli/blob/2.8.11/pyproject.toml#L30-L40
+    # They have been expanded to work with the available conda-forge packages on
+    # more recent versions of Python
+    # - colorama >=0.2.5,<0.4.4
+    - colorama >=0.2.5,<0.5.0
+    # - docutils >=0.10,<0.16
+    - docutils >=0.10,<0.18
+    - cryptography >=3.3.2,<=38.0.1
+    - ruamel.yaml >=0.15.0,<=0.17.21
+    - wcwidth <0.2.0
+    - prompt_toolkit >=3.0.24,<3.0.29
+    - distro >=1.5.0,<1.6.0
+    - awscrt >=0.12.4,<=0.14.0
+    - python-dateutil >=2.1,<3.0.0
+    - jmespath >=0.7.1,<1.1.0
+    - urllib3 >=1.25.4,<1.27
 
 test:
+  commands:
+    - aws --version
+    - aws_completer
+    # Check autocomplete index exists:
+    - python -c 'import awscli, os; assert os.stat(os.path.join(os.path.dirname(awscli.__file__), "data", "ac.index")).st_size > 1e6'
+    - COMP_LINE="aws ec2 describe-inst" aws_completer     # [unix]
   imports:
+    # find $CONDA_PREFIX/lib/python3.9/site-packages/awscli/ -type f -name __init__.py | sed -re 's%.+(awscli.*)/__init__.py%- \1%' -re s'%/%.%g' | sort
+    # Last updated version 2.8.11
     - awscli
+    - awscli.autocomplete
+    - awscli.autocomplete.local
+    - awscli.autocomplete.serverside
+    - awscli.autocomplete.serverside.custom_completers
+    - awscli.autocomplete.serverside.custom_completers.ddb
+    - awscli.autocomplete.serverside.custom_completers.logs
+    - awscli.autoprompt
+    - awscli.bcdoc
+    - awscli.botocore
+    - awscli.botocore.crt
+    - awscli.botocore.docs
+    - awscli.botocore.docs.bcdoc
+    - awscli.botocore.retries
+    - awscli.botocore.vendored
+    - awscli.botocore.vendored.requests
+    - awscli.botocore.vendored.requests.packages
+    - awscli.botocore.vendored.requests.packages.urllib3
     - awscli.customizations
+    - awscli.customizations.cloudformation
     - awscli.customizations.cloudtrail
+    - awscli.customizations.codeartifact
     - awscli.customizations.codedeploy
     - awscli.customizations.configservice
     - awscli.customizations.configure
     - awscli.customizations.datapipeline
+    - awscli.customizations.dlm
+    - awscli.customizations.dynamodb
     - awscli.customizations.ec2
+    - awscli.customizations.ecs
+    - awscli.customizations.eks
     - awscli.customizations.emr
+    - awscli.customizations.emrcontainers
     - awscli.customizations.gamelift
+    - awscli.customizations.history
+    - awscli.customizations.lightsail
+    - awscli.customizations.logs
     - awscli.customizations.s3
     - awscli.customizations.s3.syncstrategy
-  commands:
-    - test -f $PREFIX/bin/aws_completer                   # [unix]
```

Comment:
any reason for changing this to not use test or look for a specific location any more?

Reply:
I changed this to run the `aws_completer` command instead of checking for the location, since it can also be used with powershell on Windows. I rearranged the test/commands blocks so it looks like it was deleted, I've put it back to the original order so hopefully it's clearer!